flavel_game_noah.py

```diff
@@ -157,131 +145,196 @@ def see_me(self):
 
 #---------------- GAMEPLAY FUNCTIONS ----------------------#
 
-def input_check_and_route(i):
-	"""Checks input against valid commands and routes to gameplay functions"""
-	if "help" in i:
+def get_input_until_valid():
+	"""Gets input and checks input against valid commands and routes to gameplay functions"""
+	#print get_all_obj_keys()
+	print "\n",
+	i = raw_input("> ").lower()
+	print "\n",
+
+	# first we want to check some anywhere & always valid commands: help, inventory, & look around
+	if i.strip() == "help":
 		print "Here are the valid commands:"
 		print "go to - travels to a specified location"
 		print "take - takes an item if it is nearby and able to be taken"
 		print "use - uses an item if it's in your inventory and can be used"
-		print "look around - displays nearby locations you can travel to"
-		return False
+		print "look around - displays nearby locations and things of interest around you"
+		get_input_until_valid()
 
-	elif "inventory" in i:
+	elif i.strip() == "inventory":
```

Comment:
You've got a lot of 'magic strings' in here which means that if you mistype one you'll have bugs which are quiet and hard to find. Try using constants for these so you have autocomplete and noisy errors if you type invantory by mistake